src/material-examples/cdk-tree-flat/cdk-tree-flat-example.ts

```diff
@@ -117,33 +53,27 @@ export class FileDatabase {
   selector: 'cdk-tree-flat-example',
   templateUrl: 'cdk-tree-flat-example.html',
   styleUrls: ['cdk-tree-flat-example.css'],
-  providers: [FileDatabase]
 })
 export class CdkTreeFlatExample {
-  treeControl: FlatTreeControl<FileFlatNode>;
-  treeFlattener: MatTreeFlattener<FileNode, FileFlatNode>;
-  dataSource: MatTreeFlatDataSource<FileNode, FileFlatNode>;
-
-  constructor(database: FileDatabase) {
-    this.treeFlattener = new MatTreeFlattener(this.transformer, this._getLevel,
-      this._isExpandable, this._getChildren);
-    this.treeControl = new FlatTreeControl<FileFlatNode>(this._getLevel, this._isExpandable);
-    this.dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener);
-
-    database.dataChange.subscribe(data => {
-      this.dataSource.data = data;
-    });
+  private _transformer = (node: FoodNode, level: number) => {
```

Comment:
Don't use underscores for private members in the examples; that's just a convention for our libraries. 
(here and elsewhere)